Lib/test/test_concurrent_futures.py

```diff
@@ -379,6 +379,14 @@ def test_del_shutdown(self):
         for t in threads:
             t.join()
 
+    def test_shutdown_no_wait(self):
+        executor = futures.ThreadPoolExecutor(max_workers=5)
+        executor.map(abs, range(-5, 5))
```

Comment:
Is this sufficient to reproduce the issue or would you need some tasks which sleep() a bit?

Reply:
The aim of this test is just to make sure that the ressources are cleaned up eventually, even when calling `shutdown` with `wait=False`. The non-regression test is `test_shutdown_deadlock_pickle` which would fail on before this PR.